### DISCUSSION TITLE
use network mount as music directory

### DISCUSSION THREAD

**Discussion by sander816** [AzuraCast/AzuraCast#5733] [Q&A]
i have a high power vps with low disk space, and a low power backup vps, with a lot of disk space.
Can i use a network mounted directory inside azuracast? smb, sshfs or something like that?

**Comment by sander816** [comment]
i now have mounted my music dir with sshfs inside my media folder.
Azuracast sees all mp3, but when i want to add them to a playlist, the playlist is empty

**Comment by amacleod457** [comment]
Hello Sander816, and all interested parties:
I am testing this very possibility in a couple of test environments on
a local 10-gigabit network.
I have two Azuracast servers with Docker installation, and latest
stable release running, with a CIFS/SMB share mounted on each within
/etc/fstab on the host side.

So far, it is working out well, with one pretty major glitch. When
ever any of the servers are fully restarted, as in the host itself,
the CIFS/SMB mount fails, with a flurry of kernel messages to the
console.
I suspect this has something to do with Docker starting up and
glomming onto the directories before cifs.mount can get to its job.
I have added
[unit]
After=local-fs.target
to /etc/systemd/system/docker.service.d/override.conf to no avail.

My only workaround, which is certainly not feeseable on production
environments, is to ssh to the host system, then:
sudo cd /var/azuracast && docker-compose down && mount -a &&
docker-compose up -d

So while I apologize that I cannot specifically answer your question,
I am certainly interested in doing the same, and would be most
appreciative of anyone's assistance in getting the CIFS mounts to
happen without error on system startup before Docker swoops in.

Thanks and regards,
Adam

On 9/14/22, sander816 ***@***.***> wrote:
> i have a high power vps with low disk space, and a low power backup vps,
> with a lot of disk space.
> Can i use a network mounted directory inside azuracast? smb, sshfs or
> something like that?
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/AzuraCast/AzuraCast/discussions/5733
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>


-- 

Best Regards,
Adam MacLeod

**Comment by sander816** [comment]
just been very busy to set it up with nfs, and it worked ok.
After that i setup another station on the same server, and i wish i did not.

Now i cannot get it to work anymore. I have no idea whats going on, but when i add a nfs mounted dir indide my media folder, and add these mp3s to a playlist the playlist stays empty. It worked correctly an hour ago

**Comment by Moonbase59** [comment]
Personally, I’m using an NFS mount to my main music server on a local installation. Works without a glitch, but for the above-mentioned reasons I’d suggest using an NFS autofs mount instead. This will effectively only connect to the NFS server when needed, and disconnect after some (settable) timeout.

That should help avoiding "server gone away or restarted" problems.
